extensions/amp-video/1.0/video-wrapper.js

```diff
@@ -114,6 +126,26 @@ export function VideoWrapper({
     };
   }, [mediasession, playing, metadata, play, pause]);
 
+  useImperativeHandle(
+    ref,
+    () => ({
+      // Standard HTMLMediaElement/Element
+      play,
+      pause,
+      requestFullscreen,
+
+      // Non-standard
+      mute: () => setMuted(true),
+      unmute: () => {
+        if (userInteracted) {
+          setMuted(false);
+        }
+      },
+      userInteracted: () => setUserInteracted(true),
```

Comment:
Is this interaction inside video player? Or on the page as a whole?

Reply:
Video player. Then it works like the current AMP implementation (unmuted autoplay on action) just due to side-effects, which is nice:

![interact](https://user-images.githubusercontent.com/254946/93628052-c09cc200-f99a-11ea-9642-31a89225cbd3.gif)

Reply:
Then why should the `userInteracted()` method be a part of the element's imperative API? Sounds like it's then an internal state that the video adapter can derive via other APIs and events?

Reply:
Discussed offline, tracking in #30330